### PR TITLE
skip tests that depend on nonexistent or disabled models

### PR DESCRIPTION
**Pull request by drewbanin**
Fixes https://github.com/fishtown-analytics/dbt/issues/616

### CODE REVIEW

File: dbt/parser.py [hunk 1]
```diff
@@ -91,11 +91,16 @@ def process_refs(flat_graph, current_project):
...
             if target_model is None:
                 dbt.exceptions.ref_target_not_found(
                     node,
                     target_model_name,
-                    target_model_package)
+                    target_model_package,
+                    warn=node_is_test)
+
+                if node_is_test:
+                    continue
```
Comment (cmcarthur) [inline dbt/parser.py]: this doesn't seem right to me... these `dbt.exceptions` calls should raise. i'd pull the message out of `ref_target_not_found` and then make this more explicit:

```
if target_model is None:
    if node.get('resource_type') == NodeType.Test:
        logger.info(warning)
        continue
    else:
        dbt.exceptions.ref_target_not_found( ... )
```

File: dbt/parser.py [hunk 2]
```diff
@@ -92,17 +92,13 @@ def process_refs(flat_graph, current_project):
...
-                    node,
-                    target_model_name,
-                    target_model_package)
-
-            if (dbt.utils.is_enabled(node) and not
-                    dbt.utils.is_enabled(target_model)):
-                if dbt.utils.is_type(node, NodeType.Model):
-                    dbt.exceptions.ref_disabled_dependency(node, target_model)
-                else:
-                    node.get('config', {})['enabled'] = False
```
Comment (cmcarthur) [inline dbt/parser.py]: did you mean to remove this? there are two things here, ref target not found (the thing doesn't exist) and ref disabled dependency (target is disabled). i think ref target not found should still be a thing -- did functional testing prove that not to be the case?

Comment (drewbanin) [reply dbt/parser.py]: @cmcarthur yeah -- this code doesn't do anything currently. Disabled nodes are not included in the graph as of https://github.com/fishtown-analytics/dbt/pull/568.

I can leave it in, but I'm pretty sure it's dead code

Comment (cmcarthur) [reply dbt/parser.py]: gotcha

**Pull request merged**